backends-clickhouse/src/main/scala/io/glutenproject/backendsapi/clickhouse/CHTransformerApi.scala

```diff
@@ -122,20 +122,35 @@ class CHTransformerApi extends TransformerApi with Logging {
       val offHeapSize =
         nativeConfMap.getOrDefault("spark.gluten.memory.offHeap.size.in.bytes", "0").toLong
       if (offHeapSize > 0) {
-        val maxBytesBeforeExteralSort = offHeapSize * 0.5
-        nativeConfMap.put(
-          settingPrefix + "max_bytes_before_external_sort",
-          maxBytesBeforeExteralSort.toLong.toString)
-
-        val maxBytesBeforeExternalGroupBy = offHeapSize * 0.5
-        nativeConfMap.put(
-          settingPrefix + "max_bytes_before_external_group_by",
-          maxBytesBeforeExternalGroupBy.toLong.toString)
+        // Only set default max_bytes_before_external_sort for CH when it is not set explicitly.
+        val sortSpillKey = settingPrefix + "max_bytes_before_external_sort";
+        if (!nativeConfMap.containsKey(sortSpillKey)) {
+          val sortSpillValue = offHeapSize * 0.5
+          nativeConfMap.put(sortSpillKey, sortSpillValue.toLong.toString)
+        }
+
+        // Only set default max_bytes_before_external_group_by for CH when it is not set explicitly.
+        val groupBySpillKey = settingPrefix + "max_bytes_before_external_group_by";
+        if (!nativeConfMap.containsKey(groupBySpillKey)) {
+          val groupBySpillValue = offHeapSize * 0.5
+          nativeConfMap.put(groupBySpillKey, groupBySpillValue.toLong.toString)
+        }
+
+        val joinAlgorithmKey = settingPrefix + "join_algorithm";
+        if (!nativeConfMap.containsKey(joinAlgorithmKey)) {
+          nativeConfMap.put(joinAlgorithmKey, "grace_hash")
```

Comment:
we need review perforamnce

Reply:
It is a default settings to make sure OOM do not happen while executing join. Spark users could override default settings if they want to try other join algorithms.

Reply:
default settings of join_algorithm = grace_join already removed.